puma/var_vs_var.py

```diff
@@ -0,0 +1,409 @@
+        self.y_var_std = np.array(y_var_std)
+        self.key = key
+        self.fill = fill
+
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            return (
+                np.all(self.x_var_mean == other.x_var_mean)
+                and np.all(self.y_var_mean == other.y_var_mean)
+                and np.all(self.y_var_std == other.y_var_std)
+                and self.key == other.key
+            )
+        return False
+
+    def divide(self, other, inverse: bool = False):
+        """Calculate ratio between two class objects.
+
+        Parameters
+        ----------
+        other : var_vs_var class
+            Second var_vs_var object to calculate ratio with
+        inverse : bool
+            If False the ratio is calculated `this / other`,
+            if True the inverse is calculated
+
+        Returns
+        -------
+        np.ndarray
+            Ratio
+        np.ndarray
+            Ratio error
+
+        Raises
+        ------
+        ValueError
+            If binning is not identical between 2 objects
+        """
+        if not np.array_equal(self.x_var_mean, other.x_var_mean):
+            raise ValueError("The x variables of the two given objects do not match.")
+        # TODO: python 3.10 switch to cases syntax
+        nom, nom_err = self.y_var_mean, self.y_var_std
+        denom, denom_err = other.y_var_mean, other.y_var_std
+
+        ratio, ratio_err = hist_ratio(
+            denom if inverse else nom,
+            nom if inverse else denom,
+            denom_err if inverse else nom_err,
+            nom_err if inverse else denom_err,
+            step=False,
+        )
+        return (ratio, ratio_err)
+
+
+class VarVsVarPlot(PlotBase):  # pylint: disable=too-many-instance-attributes
+    """var_vs_eff plot class"""
+
+    def __init__(self, grid: bool = False, **kwargs) -> None:
+        """var_vs_eff plot properties
+
+        Parameters
+        ----------
+        grid : bool, optional
+            Set the grid for the plots.
+        **kwargs : kwargs
+            Keyword arguments from `puma.PlotObject`
+
+        Raises
+        ------
+        ValueError
+            If incompatible mode given or more than 1 ratio panel requested
+        """
+        super().__init__(grid=grid, **kwargs)
+
+        self.plot_objects = {}
+        self.add_order = []
+        self.ratios_objects = {}
+        self.reference_object = None
+        self.x_var_min = np.inf
+        self.x_var_max = -np.inf
+        self.inverse_cut = False
+        if self.n_ratio_panels > 1:
+            raise ValueError("Not more than one ratio panel supported.")
+        self.initialise_figure()
+
+    def add(
+        self, curve: object, key: str = None, reference: bool = False
+    ):  # pylint: disable=too-many-branches
+        """Adding var_vs_eff object to figure.
+
+        Parameters
+        ----------
+        curve : var_vs_eff class
+            Var_vs_eff curve
```

Comment:
I think best to refer to the class name, rather than the module name in the code (e.g. I would use `VarVsEff`), and keep the capitalisation consistent

Reply:
Fixed